client/components/Editor/plugins/inputRules.ts

```diff
@@ -104,31 +104,50 @@ const inlineMathRule = (
 // textblock starting with two dollar signs into a math block.
 const blockMathRule = (nodeType) => makeBlockMathInputRule(REGEX_BLOCK_MATH_DOLLARS, nodeType);
 
-const HTTP_MAILTO_REGEX = new RegExp(
-	// eslint-disable-next-line no-useless-escape
-	/(?:(?:(https|http|ftp)+):\/\/)?(?:\S+(?::\S*)?(@))?(?:(?:([a-z0-9][a-z0-9\-]*)?[a-z0-9]+)(?:\.(?:[a-z0-9\-])*[a-z0-9]+)*(?:\.(?:[a-z]{2,})(:\d{1,5})?))(?:\/[^\s]*)?\s$/,
-);
+const emailOrUriRegexBase =
+	'(?<emailOrUri>(?:(?:(https|http|ftp)+)://)?(?:\\S+(?::\\S*)?(?<atSign>@))?(?:(?:([a-z0-9][a-z0-9-]*)?[a-z0-9]+)(?:\\.(?:[a-z0-9-])*[a-z0-9]+)*(?:\\.(?:[a-z]{2,})(:\\d{1,5})?))(?:/[^\\s]*)?)';
 
+// Export a version of the regex and handler so that we can reuse this logic in a custom command
+// mapped to the `enter` key, because input rules don't work across nodes
+export const EMAIL_OR_URI_REGEX = new RegExp(`${emailOrUriRegexBase}$`);
+
+const EMAIL_OR_URI_REGEX_WITH_SPACE = new RegExp(`${emailOrUriRegexBase}(?<whitespace>\\s)$`);
+
+// Determine if the matched content is a url or an email and add a link mark to it
+export const linkRuleHandler = (
+	markType: MarkType,
+	appendWhitespace = false,
+	transaction?: Transaction,
+) => {
+	return (state: EditorState, match: RegExpMatchArray, start: number, end: number) => {
+		const resolvedStart = state.doc.resolve(start);
+		const tr = transaction ?? state.tr;
+		if (!resolvedStart.parent.type.allowsMarkType(markType)) {
+			return tr;
+		}
+		const emailOrUri = match.groups!.emailOrUri;
+
+		const href = `${match.groups!.atSign ? 'mailto:' : ''}${emailOrUri}`;
+
+		const link = state.schema.text(emailOrUri, [state.schema.mark(markType, { href })]);
```

Comment:
Not a regression but a quick enhancement, if this link is not prepended with https:// beforehand it to be afterwards. otherwise the link will route to a page not found.

Reply:
Same as above, I'd like to make an issue. Also, I've noticed this too. It's way too easy to create a broken link of the form `pubpub.org/pub/<pubid>/<absolute url>`. I wonder if our links should default to interpreting the URL as absolute rather than relative when there's no scheme or leading `[.]/`, but that feels like a big default change to make! Also I definitely want to make sure it's consistent across all the ways of creating a link, which feels outside the scope of this (already too-complicated) fix.